pip/commands/list.py

```diff
@@ -169,6 +182,15 @@ def get_uptodate(self, packages, options):
             if dist.latest_version == dist.parsed_version
         ]
 
+    def get_nodeps_only(self, packages, options):
+        installed_pkgs = [
+            dist for dist in self.iter_packages_latest_infos(packages, options)
```

Comment:
I'm on mobile so hard to check but I think you should not be calling `iter_packages_latest_infos` here.
You should only be filtering based on dependencies.